### ISSUE TITLE
Failing Tests

### ISSUE THREAD

**Issue by seanstrom** [request/request#1148]
I have some failing tests when I run npm test

This is the output:

```
ok test-https.js .................. 20/20
ok test-isUrl.js .................. 15/15
not ok test-localAddress.js ......... 5/6
# bind to invalid address
ok 1 should not be equal
ok 2 should be equal
ok 3 should be equal
# bind to local address
ok 4 should not be equal
not ok 5 should be equal
ok 6 should be equal

Assert: "should be equal",
         found: undefined
        wanted: undefined
        undefined:undefined

ok test-node-debug.js ............. 16/16
ok test-oauth.js .................... 7/7
ok test-onelineproxy.js ............. 3/3
ok test-option-reuse.js ........... 10/10
```

 test-localAddress.js file

``` javascript
var request = require('../index')
  , tape = require('tape')

// ...

tape('bind to local address', function(t) {
  request.get({
    uri: 'http://www.google.com',
    localAddress: '127.0.0.1'
  }, function(err, res) {
    t.notEqual(err, null)
    t.equal(err.message, 'connect EINVAL') // seems to be failing on this.
    t.equal(res, undefined)
    t.end()
  })
})
```

I noted which test is failing for me, but I'm not sure why part of my output says it expected `undefined` and got `undefined`

Also if I `console.log` out the error above the test, or add a line comment,
then it actually outputs that it got the wrong string, other wise it doesn't.
That shouldn't be happening.

In conclusion I am not sure why I don't get back the real error each time or that I get back a bad assertion error when it should assert correctly.

Any help with this would be appreciated.
